shared/hydrogen-vm-render-script.js

```diff
@@ -64,7 +64,10 @@ async function mountHydrogen() {
   const navigation = createNavigation();
   platform.setNavigation(navigation);
 
-  const archiveHistory = new ArchiveHistory(`#/session/123/room/${roomData.id}`);
+  const archiveHistory = new ArchiveHistory(
+    // We just have to match what Hydrogen is doing here.
+    `#/session/123/room/${encodeURIComponent(roomData.id)}`
```

Comment:
We have to `encodeURIComponent(roomId)` because Hydrogen was updated to do this, see https://github.com/vector-im/hydrogen-web/pull/939/files